gcc/rust/typecheck/rust-hir-type-check-type.h

```diff
@@ -25,6 +25,42 @@
 namespace Rust {
 namespace Resolver {
 
+class ArrayCapacityConstant : public TypeCheckBase
```

Comment:
this will need to change when we add in size types. It seems rust needs a constant when declaring an array type.

Reply:
By constant do you mean a constant expression (i.e. an expression that evaluates to a constant, like C++ constexpr) or a constant itself? Because I believe the requirement is for a constant expression, meaning that, for example, a call to a constant function could be listed as the array size. 
This is at least what the spec page [here](https://doc.rust-lang.org/reference/types/array.html) says.

Reply:
That is interesting I was testing this with the rust compiler:

```
fn main() {
    let x = 3;
    let arr: [i32; x];
}
```

But this errors with:

```
error[E0435]: attempt to use a non-constant value in a constant
 --> test.rs:3:20
  |
3 |     let arr: [i32; x];
  |                    ^ non-constant value

error: aborting due to previous error

For more information about this error, try `rustc --explain E0435````

Reply:
I think grammar wise its probably an expression for the capacity but it wants a constant for the static analysis.